packages/consumption/src/modules/requests/outgoing/OutgoingRequestsController.ts

```diff
@@ -105,6 +129,95 @@ export class OutgoingRequestsController extends ConsumptionBaseController {
         return result;
     }
 
+    private static validateKeyUniquenessOfRelationshipAttributesWithinRequest(items: (RequestItem | RequestItemGroup)[], recipient?: CoreAddress) {
+        const fragmentsOfMustBeAcceptedItemsOfRequest: RelationshipAttributeFragment[] = [];
+        for (const item of items) {
+            if (item instanceof RequestItemGroup) {
+                const fragmentsOfMustBeAcceptedItemsOfGroup = OutgoingRequestsController.extractRelationshipAttributeFragmentsFromMustBeAcceptedItemsOfGroup(item, recipient);
+                if (fragmentsOfMustBeAcceptedItemsOfGroup) fragmentsOfMustBeAcceptedItemsOfRequest.push(...fragmentsOfMustBeAcceptedItemsOfGroup);
+            } else {
+                const fragmentOfMustBeAcceptedRequestItem = OutgoingRequestsController.extractRelationshipAttributeFragmentFromMustBeAcceptedRequestItem(item, recipient);
+                if (fragmentOfMustBeAcceptedRequestItem) fragmentsOfMustBeAcceptedItemsOfRequest.push(fragmentOfMustBeAcceptedRequestItem);
+            }
+        }
+
+        if (OutgoingRequestsController.containsDuplicateRelationshipAttributeFragments(fragmentsOfMustBeAcceptedItemsOfRequest)) {
+            return ValidationResult.error(
+                ConsumptionCoreErrors.requests.violatedKeyUniquenessOfRelationshipAttributes(
+                    "The Request cannot be created because its acceptance would lead to the creation of more than one RelationshipAttribute in the context of this Relationship with the same key."
+                )
+            );
+        }
+
+        return ValidationResult.success();
+    }
+
+    private static extractRelationshipAttributeFragmentsFromMustBeAcceptedItemsOfGroup(requestItemGroup: RequestItemGroup, recipient?: CoreAddress) {
+        const fragmentsOfMustBeAcceptedItemsOfGroup: RelationshipAttributeFragment[] = [];
+
+        for (const item of requestItemGroup.items) {
+            if (item instanceof RequestItemGroup) {
```

Comment:
`item` is a `RequestItemDerivation`, and `RequestItemGroup` is not given as a type there, so I don't think that repeated groupings are possible

Reply:
It is correct that these lines relating to nested RequestItemGroups cannot currently be reached.  However, because we already have logic regarding nested RequestItemGroups in other places, I have also inserted it consistently (which is debatable). I think it would be a separate issue to talk about nested RequestItemGroups and if we should delete such logic. :D